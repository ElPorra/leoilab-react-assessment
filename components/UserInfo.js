import React from 'react';
import validators from '../helpers/Validators';
import validate from 'validate.js';
import {
    StyleSheet,
    Image,
    FlatList,
    View,
    Text

} from 'react-native';

export default class UserInfo extends React.Component {
    render() {
        return (
            <View style={styles.columnContainer}>
                <View style={styles.container}>
                    <Image
                        style={{ width: 50, height: 50 }}
                        source={{ uri: this.props.user.avatar_url }}
                    />
                    <Text style={styles.name}>
                        {this.props.user.name}
                    </Text>
                </View>
                <Text style={styles.name}>
                   UserId:  {this.props.user.id}
                </Text>
                </View>
        );
    }

}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'row',
        paddingTop: 22
    },
    columnContainer: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'center',
    },
    name: {
        padding: 10,
        fontSize: 18,
        height: 44,
    },
})