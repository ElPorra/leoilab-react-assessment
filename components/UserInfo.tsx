import React from "react";
import {
  StyleSheet,
  Image,
  View,
  Text,
  Modal,
  TouchableHighlight
} from "react-native";
import Layout from "../constants/Layout";
import { Icon } from "expo";
import UsersList from "./UsersList";
import ReposList from "./ReposList";

export default class UserInfo extends React.Component<
  IUserInfoProps,
  TUserInfoState
> {
  constructor(props) {
    super(props);
    this.state = {
      modalVisible: false,
      showFollowers: false,
      showRepos: false
    };
  }
  setModalVisible = (visible: boolean, whatToShow: string) => {
    let showRepos = false;
    let showFollowers = false;
    switch (whatToShow) {
      case "repos":
        showRepos = true;
        break;
      case "followers":
        showFollowers = true;
        break;
    }
    this.setState({
      modalVisible: visible,
      showRepos: showRepos,
      showFollowers: showFollowers
    });
  }

  _onCloseModal = () => {
    //
  }
  render() {
    return (
      <View style={styles.rowContainer}>
        <Image
          style={styles.image}
          source={{ uri: this.props.user.avatar_url }}
        />
        <View style={styles.columnContainer}>
          <Text style={styles.name}>{this.props.user.name}</Text>
          {this.props.repos ? (
            <TouchableHighlight
              onPress={() => {
                this.setModalVisible(!this.state.modalVisible, "repos");
              }}
            >
              <Text style={styles.name}>
                Repos: {this.props.repos.length}
                <Icon.Ionicons
                  name="md-information-circle"
                  size={Layout.sizingUnit * 2}
                  style={{ marginLeft: Layout.sizingUnit }}
                />
              </Text>
            </TouchableHighlight>
          ) : null}
          {this.props.followers ? (
            <TouchableHighlight
              onPress={() => {
                this.setModalVisible(!this.state.modalVisible, "followers");
              }}
            >
              <Text style={styles.name}>
                Followers: {this.props.followers.length}
                <Icon.Ionicons
                  name="md-information-circle"
                  size={Layout.sizingUnit * 2}
                  style={{ marginLeft: Layout.sizingUnit }}
                />
              </Text>
            </TouchableHighlight>
          ) : null}
          <Text style={styles.name}>UserId: {this.props.user.id}</Text>
        </View>
        <Modal
          animationType="slide"
          transparent={false}
          visible={this.state.modalVisible}
          onRequestClose={this._onCloseModal}
        >
          <View style={styles.modalContainer}>
            <View>
              <TouchableHighlight
                onPress={() => {
                  this.setModalVisible(!this.state.modalVisible, "");
                }}
              >
                <Icon.Ionicons
                  name="md-close-circle"
                  size={Layout.sizingUnit * 3}
                  style={{ marginLeft: Layout.sizingUnit }}
                />
              </TouchableHighlight>
             {this.state.showFollowers ? <UsersList users={this.props.followers}></UsersList>: null}
             {this.state.showRepos ?  <ReposList repos={this.props.repos}></ReposList>: null}
            </View>
          
          </View>
        </Modal>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  rowContainer: {
    flex: 1,
    flexDirection: "row",
    paddingTop: Layout.sizingUnit
  },
  columnContainer: {
    flex: 1,
    flexDirection: "column",
    justifyContent: "center"
  },
  name: {
    padding: Layout.sizingUnit,
    fontSize: Layout.sizingUnit * 1.8,
    height: Layout.sizingUnit * 4.4
  },
  image: { width: Layout.sizingUnit * 10, height: Layout.sizingUnit * 10 },
  modalContainer: { marginTop: Layout.sizingUnit * 2 }
});
