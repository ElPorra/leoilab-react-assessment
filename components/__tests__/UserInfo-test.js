import 'react-native';
import React from 'react';
import UserInfo from '../UserInfo';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const userData = {
        "login": "adegano",
        "id": "29756043",
        "avatar_url": "https://avatars1.githubusercontent.com/u/29756043?v=4",
        "name": "adegano",
      }
  const tree = renderer.create(<UserInfo user = {userData}></UserInfo>).toJSON();

  expect(tree).toMatchSnapshot();
});
