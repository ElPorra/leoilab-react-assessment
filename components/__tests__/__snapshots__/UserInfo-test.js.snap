// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingTop": 22,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://avatars1.githubusercontent.com/u/29756043?v=4",
        }
      }
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 18,
          "height": 44,
          "padding": 10,
        }
      }
    >
      adegano
    </Text>
  </View>
  <Text
    style={
      Object {
        "fontSize": 18,
        "height": 44,
        "padding": 10,
      }
    }
  >
    UserId:  
    29756043
  </Text>
</View>
`;
